SwanCustomEnvironments/swancustomenvironments/serverextension.py

```diff
@@ -36,17 +38,31 @@ async def get(self):
             # The first get request will launch the execution of the makenv.sh script,
             # to create the environment
             makenv_process = self._launch_makenv()
-            SwanCustomEnvironmentsApiHandler.makenv_process = makenv_process
             log_file = open(self.LOG_FILE, "w")
+            SwanCustomEnvironmentsApiHandler.makenv_process = makenv_process
+            SwanCustomEnvironmentsApiHandler.writing_log = True
```

Comment:
I'm thinking that it would be clearer if we place this code (and what is after the if else) in a helper method called `_process_first_get`, and what is in the else part in another helper method called `_process_subsequent_get`.